unitTests/CommunicationPattern.C

```diff
@@ -24,75 +24,264 @@ protected:
 };
 
 
-TEST_F(CommunicationPatternFixture, can_get_comm_size)
+TEST_F(CommunicationPatternFixture, compute_owner_rank_single_owner)
 {
+    // Arrange
     auto comm = exec->get_gko_mpi_host_comm();
-    EXPECT_EQ(comm->size(), 4);
+
+    // Act
+    auto owner_rank = compute_owner_rank(comm->rank(), comm->size());
+
+    // Assert
+    EXPECT_EQ(owner_rank, 0);
 }
 
-TEST_F(CommunicationPatternFixture, compute_owner_rank)
+TEST_F(CommunicationPatternFixture, compute_owner_rank_two_owners)
 {
+    // Arrange
     auto comm = exec->get_gko_mpi_host_comm();
-    auto owner_rank = compute_owner_rank(comm->rank(), comm->size());
-    // if ranks_per_owner is same as total number of ranks all ranks have
-    // zero as owner
-    EXPECT_EQ(owner_rank, 0);
+    auto comm_rank = comm->rank();
+    auto comm_size = comm->size();
+
+    // Act
+    auto owner_rank = compute_owner_rank(comm_rank, comm_size/2);
+
+    // Assert
+    if (comm_rank < comm_size/2)
+        EXPECT_EQ(owner_rank, 0);
+    else
+        EXPECT_EQ(owner_rank, comm_size/2);
 }
 
-TEST_F(CommunicationPatternFixture, compute_gather_to_owner_counts_all_owner)
+TEST_F(CommunicationPatternFixture, compute_owner_rank_all_owners)
 {
+    // Arrange
     auto comm = exec->get_gko_mpi_host_comm();
+    auto comm_rank = comm->rank();
+
+    // Act
+    auto owner_rank = compute_owner_rank(comm_rank, 1);
+
+    // Assert
+    EXPECT_EQ(owner_rank, comm_rank);
+}
+
+TEST_F(CommunicationPatternFixture, compute_scatter_from_owner_counts_single_owner)
+{
+    // Arrange
+    auto comm = exec->get_gko_mpi_host_comm();
+    auto num_elements = 10;
+    auto comm_size = comm->size();
+    auto comm_rank = comm->rank();
+
+    // Scatter different number of elements from owner to each non-owner process
+    std::vector<int> send_counts(comm_size);
+    
+    if (comm_rank == 0)
+        for (int i = 0; i < comm_size; i++)
+            send_counts[i] = num_elements*i;
```

Comment:
Please add curly braces here, even if it is correct notation without. 
```suggestion
        for (int i = 0; i < comm_size; i++) {
            send_counts[i] = num_elements*i;
            }
```